src/core/main.go

```diff
@@ -202,9 +206,23 @@ func main() {
 	if err := dao.InitDatabase(database); err != nil {
 		log.Fatalf("failed to initialize database: %v", err)
 	}
-	if err = migration.Migrate(database); err != nil {
-		log.Fatalf("failed to migrate: %v", err)
+	if strings.EqualFold(*runMode, "migrate") {
+		// Used by Harbor helm preinstall, preupgrade hook container
+		if err = migration.Migrate(database); err != nil {
+			log.Fatalf("failed to migrate the database, error: %v", err)
+			os.Exit(1)
```

Comment:
log.Fatalf calls the os.Exit(1) eventually, please remove it.

Reply:
updated